### PR TITLE
feat: add TC migration option

### PR DESCRIPTION
**Pull request by csatib02**
## Overview

- The core addition of this PR is to enable log collection by utilizing the [Telemetry-Controller](https://github.com/kube-logging/telemetry-controller) resources. This is done via the `logging` resource.
- There is a new field on `logging` that will determine how TC resources will be created:

```yaml
apiVersion: logging.banzaicloud.io/v1beta1
kind: Logging
metadata:
  name: tc-config
spec:
  routeConfig:
    enableTelemetryControllerRoute: true # Determines whether to create TC resources for log-collection and routing purposes.
    disableLoggingRoute: false # Determines whether to use Logging-routes for routing purposes and Fluentbit-agents for log-collection.
    tenantLabels: # Will be placed on TC's tenant resources, must be matched with the same field on the Collector resource. (Deployed by the cluster administrator.)
      tenant: logging
```
- To use this one most set the `enable-telemetry-controller-route` flag to `true` Eg.:
```sh
helm install logging-operator ./charts/logging-operator/ --set extraArgs='{"-enable-leader-election=true","-enable-telemetry-controller-route"}' --set telemetry-controller.install=true
```

- Adds option to install telemetry-controller as a sub-chart.
- Sets all API occurrences of the `ControlNamespace` field to be immutable.

### CODE REVIEW

File: pkg/sdk/logging/api/v1beta1/logging_types.go [hunk 1]
```diff
@@ -114,6 +121,19 @@ type ConfigCheck struct {
 	Labels map[string]string `json:"labels,omitempty"`
 }
 
+type RouteConfig struct {
+	// If DisableLoggingRoute is set to true, the logging route controller
+	// should remove the given tenant from the status of the logging resource.
+	DisableLoggingRoute bool `json:"disableLoggingRoute,omitempty"`
+	// If EnableTelemtryControllerRoute set to true, the opoerator will create
```
Comment (OverOrion) [inline pkg/sdk/logging/api/v1beta1/logging_types.go:128]:
```suggestion
	// If EnableTelemtryControllerRoute set to true, the operator will create
```

File: pkg/sdk/logging/api/v1beta1/logging_types.go [hunk 2]
```diff
@@ -92,6 +96,9 @@ type LoggingSpec struct {
 	// - downstream parsers can use the `log` field instead of `message` as they did with the docker runtime
 	// - the `concat` and `parser` filters are automatically set back to use the `log` field
 	EnableDockerParserCompatibilityForCRI bool `json:"enableDockerParserCompatibilityForCRI,omitempty"`
+	// RouteConfig determines whether to use loggingRoutes or to create resources based on the logging resource
+	// that can be managed by the Telmetry Controller.
```
Comment (OverOrion) [inline pkg/sdk/logging/api/v1beta1/logging_types.go:100]:
```suggestion
	// that can be managed by the Telemetry Controller.
```

**Review by OverOrion — COMMENTED** [review]
Besides 2 typos this looks good to me, nice! :rocket:

File: pkg/sdk/logging/api/v1beta1/logging_types.go [hunk 3]
```diff
@@ -114,6 +121,19 @@ type ConfigCheck struct {
...
+	// If DisableLoggingRoute is set to true, the logging route controller
+	// should remove the given tenant from the status of the logging resource.
+	DisableLoggingRoute bool `json:"disableLoggingRoute,omitempty"`
+	// If EnableTelemtryControllerRoute set to true, the operator will create
+	// the corresponding Tenant, Subscription, Output based on the logging resource.
+	EnableTelemetryControllerRoute bool `json:"enableTelemetryControllerRoute,omitempty"`
+	// TenantLabels is a map of labels that will be added to the tenant object
+	// so it can be matched with TelemetryController's TenantSelector
+	// ref: https://github.com/kube-logging/telemetry-controller/blob/main/api/telemetry/v1alpha1/collector_types.go
+	TenantLabels *metav1.LabelSelector `json:"tenantLabels,omitempty"`
```
Comment (csatib02) [inline pkg/sdk/logging/api/v1beta1/logging_types.go:134]: We might want to change this to a simple `map[string]string`
@OverOrion @pepov WDYT?

**Review by OverOrion — APPROVED** [review]

**Pull request merged**